site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+child to that parent.
+
+### Failed Binding
+
+The binding transaction MUST fail if there is A non-recoverable error during the
+binding process. A failed binding should result in no changes to the
+configuration of the active Ephemeral route.
+
+A failed binding DOES NOT invalidate the parent `RouteRules` that the child
+route is trying to bind to. This is necessary because other child routes might
+bind to those parental `RouteRules` (might already be bound or may try to bind
+in the future).
+
+When a parent route's binding to a child route fails, that should not block the
+parent route's bindings with other child routes. In the same way, a child route
+failing to bind with a parent route should not block that child's bindings to
+other parent routes.
+
+## Reconciling Parent and Child Routes
+
+**A key principle guiding the design of how parent and child routes are
+combined, is that the data plane can apply the resulting route settings in a
+single processing cycle (per direction) of the network traffic.**
+
+NOTE: In the rest of this section, the term configuration "object" can also
+represent a config parameter or an array of either.Which of those it represents
+depends on the specific config setting it is being applied to and what the
+containing object supports.
+
+Conceptually, the basic the way a parent `Route` and child `Routes` are combined
+is that, an object in the parent route is added (logical AND) to **each**
+equivalent object(s) in the child route. This starts at the top layer of objects
+and is repeated, in sequence, down each layer.
+
+For any given level of config objects there may be a 1-to-many relationship. As
+an example, the settings in the parent's`RouteRule` are "added" to each of the
+child's `RouteRule`s.
+
+### Configuration Conflicts and Invalid Configurations
+
+During the reconciliation process, any conflict of configuration settings
+between the parent and child routes is considered fatal to the transaction and
+the entire Binding fails.
+
+A configuration conflict exists when the reconciliation process would result in
+an Ephemeral route configuration that would not be valid as a single, standalone
+`Route`.  This means, in concept, a user could save the generated Ephemeral
+route config to `my-foo-route.yaml`, remove the parent and child routes from the
+cluster, then load `my-foo-route.yaml` into the cluster, and the system
+would accept and deploy it as a valid config.
+
+A config conflict may not be between config elements in the child and the
+parent. Instead, the conflict may be between config elements from two (or more)
+child routes that attach to the same `RouteRule` in the parent route. For this
+reason, when a config conflict is detected, implementations should identify the
+config elements that conflict, including the routes that are the original source
+of each element, and provide that info to users via appropriate methods of
+notification  (e.g., logs, message in GUI, etc.).
```

Comment:
Should we prescribe a status message that should occur as a result? Talking about UI, etc seem to be outside of scope -- all metadata like this in the API today is via status. And extra layers on top should be outside of the spec and depend on status